src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -171,6 +174,92 @@ private byte[] GetBytesFromSlave(int count, TimeInterval timeInterval)
             return result;
         }
 
+        // When we need to read for more than 255 bytes, we will need this method
+        private byte[] GetLongBytes(int count, TimeInterval timeInterval) {
+            if (count <= 255) {
+                return GetBytesFromSlave(count, timeInterval);
+            }
```

Comment:
I think it's better to remove this condition to always use the long path, to remove a possible failure branch.

It's less performant but I think it should be more than fast enough regardless.

Reply:
Done